--- tools/rosbag/CMakeLists.txt.orig	2014-08-01 09:42:56.359487097 +0200
+++ tools/rosbag/CMakeLists.txt	2014-07-31 16:40:37.035703175 +0200
@@ -9,7 +9,7 @@ catkin_python_setup()
 # Support large bags (>2GB) on 32-bit systems
 add_definitions(-D_FILE_OFFSET_BITS=64)
 
-include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
+include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} ${BZIP2_INCLUDE_DIR})
 
 catkin_package(
   LIBRARIES rosbag
